src/v/iceberg/schema.cc

```diff
@@ -69,4 +69,23 @@ schema::ids_to_types(chunked_hash_set<nested_field::id_t> target_ids) const {
     return ret;
 }
 
+std::optional<nested_field::id_t> schema::highest_field_id() const {
+    chunked_vector<const nested_field*> to_visit;
+    for (const auto& field : schema_struct.fields) {
+        to_visit.emplace_back(field.get());
+    }
+    std::optional<nested_field::id_t> highest;
+    while (!to_visit.empty()) {
```

Comment:
Why is this done with two loops? I thought you were setting up for a depth first search, but it doesn't look like it's recursing down nested structures. It seems like all of this could be in the first `for` loop.

Reply:
Nice catch! Missed a couple lines... Fixed